lib/ecs_deploy/auto_scaler/auto_scaling_group_config.rb

```diff
@@ -63,10 +63,17 @@ def cluster_resource_manager
         )
       end
 
+      # NOTE: InstanceDrainer calls this method when it receives spot instance interruption warnings
       def detach_instances(instance_ids:, should_decrement_desired_capacity:)
         return if instance_ids.empty?
 
-        instance_ids.each_slice(MAX_DETACHABLE_INSTANCE_COUNT) do |ids|
+        # detach only detachable instances
+        auto_scaling_group_instances = describe_detachable_instances
+        detachable_instance_ids = instance_ids.select do |instance_id|
+          auto_scaling_group_instances.any? { |instance| instance.instance_id == instance_id }
+        end
```

Comment:
How about this code?
```ruby
detachable_instance_ids = instance_ids & describe_detachable_instances.map(&:instance_id)
```

Reply:
fixed in https://github.com/reproio/ecs_deploy/pull/96/commits/e397ae7d5d459ab7dcbfdb1383d9758fac5f662b